go/v1beta1/storage/esutil/index.go

```diff
@@ -0,0 +1,235 @@
+// Copyright 2021 The Rode Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package esutil
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/elastic/go-elasticsearch/v7"
+	"go.uber.org/zap"
+)
+
+const (
+	ProjectDocumentKind    = "projects"
+	OccurrenceDocumentKind = "occurrences"
+	NoteDocumentKind       = "notes"
```

Comment:
You can make your own type here to prevent unrecognized strings from being used as arguments to functions within this package.

```go
type DocumentKind string

const (
  ProjectDocumentKind    DocumentKind = "projects"
  OccurrenceDocumentKind DocumentKind = "occurrences"
  NoteDocumentKind       DocumentKind = "notes"
)
```

Then, functions / methods such as `getMappingForDocumentKind` could have a signature that looks like this:

```go
func (em *EsIndexManager) getMappingForDocumentKind(documentKind DocumentKind) (*VersionedMapping, error)
```

Which will help prevent unrecognized strings from being used here.